### PR TITLE
Notification badge for counters

### PR DESCRIPTION
**Pull request by maxmarkus**

### CODE REVIEW

**Comment by maxmarkus** [comment]
Implemented for normal buttons as well as categories in top nav. Works for desktop and mobile.
preview of implementation

File: core/src/core-api/navigation.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class LuigiNavigationManager {
+  /**
+   * Use these functions for navigation related features
+   * @name LuigiNavigation
+   */
+  constructor() {}
+
+  /**
+   * Refreshes top navigation badge counters
+   * Basically triggeres a re-render of the navigation.
```
Comment (bszwarc) [inline core/src/core-api/navigation.js]:
```suggestion
  
```
I would combine this sentence with the one above.

File: core/src/core-api/navigation.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class LuigiNavigationManager {
+  /**
+   * Use these functions for navigation related features
+   * @name LuigiNavigation
+   */
+  constructor() {}
+
+  /**
+   * Refreshes top navigation badge counters
```
Comment (bszwarc) [inline core/src/core-api/navigation.js]:
```suggestion
   * Refreshes top navigation badge counters by rendering the navigation again.
```

File: core/src/core-api/navigation.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class LuigiNavigationManager {
+  /**
+   * Use these functions for navigation related features
```
Comment (bszwarc) [inline core/src/core-api/navigation.js]:
```suggestion
   * Use these functions for navigation-related features.
```

File: docs/navigation-parameters-reference.md [hunk 4]
```diff
@@ -183,6 +183,10 @@ The node parameters are as follows:
 - **viewGroup** defines a group of views in the same domain sharing a common security context. This improves performance through reusing the frame. Use **viewGroup** only for the views that use path routing internally.
 - **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation.
 - **hideSideNav** if set to `true`, the left navigation disappears when you click the affected node. It is set to `false` by default.
+- **badgeCounter** adds a badge with a number and a label to a Node. Nodes that are part of a category will show a cumulated number of all badges in this category. **badgeCounter** is only available for top navigation items.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **badgeCounter** adds a badge with a number and a label to a node. Nodes that are part of a category show a cumulated number of all badges in this category. **badgeCounter** is only available for top navigation items.
```

File: docs/navigation-parameters-reference.md [hunk 5]
```diff
@@ -183,6 +183,10 @@ The node parameters are as follows:
...
+  - **label** is the label of the badge
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
  - **label** is the label of the badge.
```

File: docs/navigation-parameters-reference.md [hunk 6]
```diff
@@ -183,6 +183,10 @@ The node parameters are as follows:
...
+  - **count** is a function or asynchronous function that returns a number.
+  Gets updated when clicked on the navigation. Use `Luigi.navigation().updateTopNavigation()` in Luigi Core or execute `window.parent.postMessage({msg: 'luigi.navigation.update-badge-counters'}, '*');` from within a micro-frontend to trigger a manual update.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
  Gets updated when you click the navigation. Use `Luigi.navigation().updateTopNavigation()` in Luigi Core or execute `window.parent.postMessage({msg: 'luigi.navigation.update-badge-counters'}, '*');` from within a micro frontend to trigger a manual update.
```

**Pull request merged**